gpt_engineer/core/step_bundle_interface.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+from gpt_engineer.core.code import Code
+from gpt_engineer.core.ai import AI
+
+
+class StepBundleInterface(ABC):
+
+    @abstractmethod
+    def gen_code(self, ai: AI, prompt: str) -> Code:
+        pass
+
+    @abstractmethod
+    def improve_code(self, ai: AI, prompt: str) -> Code:
```

Comment:
```suggestion
    def improve(self, ai: AI, prompt: str) -> Code:
```

Reply:
Shouldn't this accept Code as an argument?

(Also, maybe we should give access to "memory"?)

Reply:
Yes, the signature of this will change and I'm thinking about making two improve classes, one that takes one Code (standard) and one that takes 2 Code (for improving a certain part). The second part will be crucial for writing larger programs.